### PR TITLE
Setup GH action

### PR DESCRIPTION
**Pull request by ewgenius**
- switched from Travis CI to actions
- added [super-linter action](https://github.com/github/super-linter)
> note: linter for now configured to run only on new/changed files - it detects quite big amount of issues at the moment, so it is better to fix them gradually

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    name: Lint Code Base
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+      - name: Lint Code Base
+        uses: docker://github/super-linter:v2.1.1
+        env:
+          VALIDATE_ALL_CODEBASE: false
+          VALIDATE_ANSIBLE: false
```
Comment (oakeyc) [inline .github/workflows/lint.yml:19]: new line ?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      
+    steps:
+    - name: Install Dependencies (Linux)
+      run: |
+        sudo apt-get update -y
+        sudo apt-get install -y devscripts debhelper moreutils fakeroot jq pigz
+        wget "https://storage.googleapis.com/shellcheck/shellcheck-v0.7.0.linux.x86_64.tar.xz"
+        tar --xz -xvf "shellcheck-v0.7.0.linux.x86_64.tar.xz"
+        sudo cp shellcheck-v0.7.0/shellcheck /usr/bin/shellcheck
+      if: matrix.os != 'macos-latest'
```
Comment (ipmsteven) [inline .github/workflows/main.yml:22]: how about 
```
if: matrix.os == 'ubuntu-latest'
```
which is consistent with the `if: matrix.os == 'macos-latest'` below

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    - name: Get Sources
+      uses: actions/checkout@v2
+    - name: Test
+      run: |
+        export PATH="$PATH:/snap/bin"
+        make test
+      shell: bash
+    - name: Build (Linux)
+      run: DEB_BUILD_OPTIONS=nocheck debuild -us -uc
+      if: matrix.os != 'macos-latest'
```
Comment (ipmsteven) [inline .github/workflows/main.yml:38]: same here  https://github.com/github/backup-utils/pull/618/files#r443738293

File: .github/workflows/lint.yml (hunk 1 above)
Comment (ipmsteven) [inline .github/workflows/lint.yml:19]: nit: add a new line at the end of this file

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      uses: actions/checkout@v2
+    - name: Test
+      run: |
+        export PATH="$PATH:/snap/bin"
+        make test
+      shell: bash
+    - name: Build (Linux)
+      run: DEB_BUILD_OPTIONS=nocheck debuild -us -uc
+      if: matrix.os != 'macos-latest'     
+
```
Comment (ipmsteven) [inline .github/workflows/main.yml:39]: nit: add a new line at the end of this file

File: .github/workflows/lint.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
+name: Lint Code Base
```
Comment (ipmsteven) [inline .github/workflows/lint.yml:1]: question: Will this block the PR or just show a warning when linter found some violation?

File: .github/workflows/lint.yml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+jobs:
+  build:
+    name: Lint Code Base
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+      - name: Lint Code Base
+        uses: docker://github/super-linter:v2.1.1
```
Comment (ipmsteven) [inline .github/workflows/lint.yml:16]: could you check we are using the same bash ruleset which we already used in enterprise2?

File: .github/workflows/main.yml (hunk 2 above)
Comment (ewgenius) [reply .github/workflows/main.yml:22]: Yeah, makes sense for now. But we will need to change it back, if will be adding older ubuntu versions in testing (once flaky test fixed for them)

File: .github/workflows/lint.yml (hunk 6 above)
Comment (ewgenius) [reply .github/workflows/lint.yml:16]: As far as I know `shellcheck` uses `#shellcheck` inline comments and `.shellcheckrc` file for single place configuration. We don't have such file in enterprise2, so I guess it falls back to inline directives and default configuration

File: .github/workflows/lint.yml (hunk 1 above)
Comment (ewgenius) [reply .github/workflows/lint.yml]: :+1:

Comment (ewgenius) [reply .github/workflows/lint.yml]: 👍

File: .github/workflows/main.yml (hunk 4 above)
Comment (ewgenius) [reply .github/workflows/main.yml]: 👍

File: .github/workflows/lint.yml (hunk 5 above)
Comment (ewgenius) [reply .github/workflows/lint.yml:1]: No, it won't:
- for now we have only Travis CI check as required. Once This PR completed, I'll switch it to build GH action, but linting can be optional

image

**Pull request merged**